fl4health/clients/fenda_client.py

```diff
@@ -144,10 +162,10 @@ def get_contrastive_loss(
         Compute contrastive over current local features (z_p) with old local features (hat{z_p}) as positive pairs
         and current global features (z_s) as negative pairs.
         """
-        return self.compute_contrastive_loss(
+        return self.contrastive_loss(
             features=local_features,  # (z_p)
-            positive_pairs=old_local_features,  # (\hat{z_p})
-            negative_pairs=global_features,  # (z_s)
+            positive_pairs=old_local_features.unsqueeze(0),  # (\hat{z_p})
```

Comment:
I may be missing something, but why do we need to inject this unsqueeze here now?

Reply:
I have implemented contrastive loss based on MOON implementation, where the negative pair of a single sample can be multiple samples based on the list of saved previous models. For the consistency in design, the contrastive loss here gets a tensor of (n_pairs ,n_samples, n_features) as negative_pairs and a tensor of (1, n_samples, n_features) as positive pairs as each sample can only have one positive pair. Such a squeeze here is to fit in these shape sizes.